backend/algorithms/objects/conditions.py

```diff
@@ -217,6 +232,13 @@ def get_warning(self, applicable_wam: Optional[float]) -> str:
             return wam_warning
         return f"{wam_warning} Your WAM in {category} is currently {applicable_wam:.3f}"
 
+    def condition_to_model(self, model: cp_model.CpModel, user: User, courses: list[Tuple[cp_model.IntVar, Course]], course_variable: cp_model.IntVar) -> list[cp_model.Constraint]:
```

Comment:
Thoughts on the way this is? I just assumed the user was loaded up with some previous courses, and we can not predict future wam, Not sure tho.